src/components/chart-preview/ChartPreview.tsx

```diff
@@ -71,6 +85,59 @@ export const ActualChart = ({
     return fullFigureNode;
   };
 
+  const onLegendClick = (e: any) => {
+    // check if the passed 'e.curveNumber' is in filteredColumnIndexes
+    // if it is, that means its filter is on and should be turned off
+    // if it isn't it means its filter should be turned on
+
+    // e.data shows what filters are applied BUT it is showing the state at time before
+    // the clicking so can't be used for current filters applied
+    // this causes more problems with onLegendDoubleClick where multiple filters can be applied
+    // and turned off on a single trigger
+
+    if (filteredColumnIndexes.includes(e.curveNumber)) {
+      const index = filteredColumnIndexes.indexOf(e.curveNumber);
+      if (index > -1) {
+        filteredColumnIndexes.splice(index, 1);
+      }
+    } else {
+      filteredColumnIndexes.push(e.curveNumber);
+    }
+    filterData(e.data, filteredColumnIndexes);
+  };
+
+  const onLegendDoubleClick = (e: any) => {
+    // a note on how plotly legend double click works
+    // when no filters are on, it'll turn all filters on except the clicked one
+    // when only one filter is not on, and any are clicked, it'll turn all filters off
+    // when two or fewer filters are not on, and the one clicked is no on it'll turn all filters on except the clicked one
+    // when two or fewer filters are not on, and the one clicked is on, it'll turn all filters off
+
+    // to keep track of what filters are applied we have an array filteredColumnIndexes
+    // this has the 'curveNumber' of each data series
+
+    let numberedArray = e.data.map((_elem: any, index: any) => index);
+    if (filteredColumnIndexes.includes(e.curveNumber)) {
+      filterData(e.data, []);
+    } else if (filteredColumnIndexes.length < e.data.length - 1) {
+      const temp = numberedArray.filter((x: any) => x !== e.curveNumber);
+      filterData(e.data, temp);
+    } else {
+      filterData(e.data, []);
+    }
+  };
+
+  const filterData = (data: any[], indexes: number[]) => {
+    const filteredData = data.filter(
+      (_element: any, index: number) => !indexes.includes(index),
+    );
+    // this setting of filteredColumnIndex is here because onLegendClick fires twice when
```

Comment:
I dont know if this is the same thing, but would configuring the double click delay help prevent it getting picked up as 2 clicks? https://plotly.com/javascript/configuration-options/#double-click-delay

Reply:
Good find, I had a play with it but it was still firing the single click twice. It was just delaying how long it took to put the filter on the actual chart, this could be an issue with plotly though.
https://plotly.com/javascript/reference/layout/#layout-legend-itemdoubleclick
We do have the option of changing what the double click does.